### ISSUE TITLE
Issue with themes installed as federated extensions

### ISSUE THREAD

**Issue by jtpio** [jupyterlab/jupyterlab#10378]
## Description

This was discussed during the JupyterLab weekly call today.

There seems to be some issues with styles not being unloaded when switching themes.

## Reproduce

There are at least two occurences of this at the moment in third-party themes:

- https://github.com/telamonian/theme-darcula/issues/32
- https://github.com/timkpaine/jupyterlab_miami_nights/pull/14#issuecomment-854453165 

![image](https://user-images.githubusercontent.com/591645/120765897-f7f7d280-c519-11eb-9763-6e65094a9289.png)

## Expected behavior

Themes should unload correctly.

## Context

cc @telamonian @krassowski

**Comment by jasongrout** [comment]
Investigating, the problem is that (a) the theme gives the css in its style key, so it is packaged into the prebuilt bundle as a webpack module, and then (b) jlab doesn't do what the documentation claims, i.e., ignore any css files if there is a themePath defined (probably my mistake in writing that doc?)

We can ignore any css file if a theme path is defined by this patch to jupyterlab_server, which fixes this issue:

```diff
diff --git a/jupyterlab_server/config.py b/jupyterlab_server/config.py
index 56e0f42..02df55c 100644
--- a/jupyterlab_server/config.py
+++ b/jupyterlab_server/config.py
@@ -128,7 +128,8 @@ def get_page_config(labextensions_path, app_settings_dir=None, logger=None):
             extension['extension'] = extbuild['extension']
         if 'mimeExtension' in extbuild:
             extension['mimeExtension'] = extbuild['mimeExtension']
-        if 'style' in extbuild:
+        # Ignore any style files if there is a theme path defined, per the documentation.
+        if 'style' in extbuild and 'themePath' not in ext_data['jupyterlab']:
             extension['style'] = extbuild['style']
         extensions.append(extension)
```
However, perhaps it is better to have themes not define their theme css as a `style` key in the package.json?

**Comment by jasongrout** [comment]
Can you try just deleting the style/styleModule keys in the darcula package.json?

With https://github.com/jupyterlab/jupyterlab/pull/10381, the default extensions delete their style/styleModule keys.

**Comment by jasongrout** [comment]
Fixed by #10381.

The main thing fixed by that PR is updating the documentation to accurately reflect that theme css should not be pointed to by the style/styleModule keys. The other part of the PR is enacting that change for our core themes.

**Comment by jasongrout** [comment]
> Can you try just deleting the style/styleModule keys in the darcula package.json?

This works today, with no changes to jlab. See https://github.com/telamonian/theme-darcula/pull/33, for example.
